lib/gpu/lib/src/texture.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+part of flutter_gpu;
+
+base class Texture extends NativeFieldWrapperClass1 {
```

Comment:
Can this be more faithful to the the C++ API with the concept of descriptors and allocators obtaining texture handles?

Reply:
Yeah I'm interested in hearing your thoughts around this. These breaks from Impeller are intentional, but indeed we'll keep tinkering and I'm super open to changing these things.

Will we ever need multiple device allocators? We ended up not using multiple for Entities, and most HALs seem to just tie allocations to the device. We did discuss the possibility of making contexts for multiple devices/having a device selection scheme in the future in the doc. There's a rough plan in case we need it, but those use cases are pretty eccentric.

The Flutter GPU doc removed a lot of descriptors from the beginning, opting for just using optional params in a lot of places instead. I think we should still have them for the more complicated config substructures like `RenderPass` attachments, but not sure we're gaining anything usability-wise by using them for texture and buffer creation?